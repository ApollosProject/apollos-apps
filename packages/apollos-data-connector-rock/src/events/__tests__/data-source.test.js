import ApollosConfig from '@apollosproject/config';
import EventsDataSource from '../data-source';

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    SHARE_URL: 'https://apollosrock.newspring.cc',
    TIMEZONE: 'America/New_York',
  },
  ROCK_MAPPINGS: {
    SERMON_CHANNEL_ID: 'TEST_ID',
  },
});

describe('Events', () => {
  it('should return start and end based on a schedule', async () => {
    const Event = new EventsDataSource();

    const result = await Event.getDateTime({
      iCalendarContent:
        'BEGIN:VCALENDAR\r\nBEGIN:VEVENT\r\nDTEND:20130501T190000\r\nDTSTART:20130501T180000\r\nRRULE:FREQ=WEEKLY;BYDAY=SA\r\nEND:VEVENT\r\nEND:VCALENDAR',
    });
    expect(result).toMatchSnapshot();
  });

  it('should return a name based on an EventItem', async () => {
    const Event = new EventsDataSource();

    Event.get = jest.fn(() =>
      Promise.resolve({
        name: 'some event',
      })
    );

    const result = await Event.getName({ eventItemId: 123 });
    expect(result).toEqual('some event');
    expect(Event.get.mock.calls).toMatchSnapshot();
  });

  it('should return a description based on an EventItem', async () => {
    const Event = new EventsDataSource();

    Event.get = jest.fn(() =>
      Promise.resolve({
        description: 'some description',
      })
    );

    const result = await Event.getDescription({ eventItemId: 123 });
    expect(result).toEqual('some description');
    expect(Event.get.mock.calls).toMatchSnapshot();
  });

  it('should return an image based on an EventItem', async () => {
    const Event = new EventsDataSource();

    Event.get = jest.fn(() =>
      Promise.resolve({
        name: 'some event',
      })
    );

    const imageUrl = jest.fn(() => Promise.resolve('https://example.com'));

    Event.context = {
      dataSources: { BinaryFiles: { findOrReturnImageUrl: imageUrl } },
    };

    const result = await Event.getImage({ eventItemId: 123 });
    expect(result).toMatchSnapshot();
    expect(Event.get.mock.calls).toMatchSnapshot();
    expect(imageUrl.mock.calls).toMatchSnapshot();
  });

  it('should get by Campus', async () => {
    const Event = new EventsDataSource();

    Event.get = jest.fn(() =>
      Promise.resolve([
        {
          id: 123,
          name: 'some event',
        },
      ])
    );

    const result = await Event.getByCampus(456);
    expect(result).toMatchSnapshot();
    expect(Event.get.mock.calls).toMatchSnapshot();
  });

  it('should get all recent events', async () => {
    const Event = new EventsDataSource();

    Event.get = jest.fn(() =>
      Promise.resolve([
        {
          id: 123,
          name: 'some event',
        },
      ])
    );

    const result = await Event.findRecent().get();
    expect(result).toMatchSnapshot();
    expect(Event.get.mock.calls).toMatchSnapshot();
  });

  it('should get all recent events using the plugin', async () => {
    ApollosConfig.loadJs({
      ROCK: { USE_PLUGIN: true },
    });
    const Event = new EventsDataSource();

    Event.get = jest.fn(() =>
      Promise.resolve([
        {
          id: 123,
          name: 'some event',
        },
      ])
    );

    const result = await Event.findRecent().get();
    expect(result).toMatchSnapshot();
    expect(Event.get.mock.calls).toMatchSnapshot();
    ApollosConfig.loadJs({
      ROCK: { USE_PLUGIN: false },
    });
  });

  it('should get from id', async () => {
    const Event = new EventsDataSource();

    Event.get = jest.fn(() =>
      Promise.resolve({
        id: 123,
        name: 'some event',
      })
    );

    const result = await Event.getFromId(123);
    expect(result).toMatchSnapshot();
    expect(Event.get.mock.calls).toMatchSnapshot();
  });
});
