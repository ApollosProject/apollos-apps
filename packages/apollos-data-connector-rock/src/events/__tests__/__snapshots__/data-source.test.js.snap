// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events should get all recent events 1`] = `
Array [
  Object {
    "id": 123,
    "name": "some event",
  },
]
`;

exports[`Events should get all recent events 2`] = `
Array [
  Array [
    "EventItemOccurrences?loadAttributes=expanded&%24expand=Schedule&%24orderby=Schedule%2FEffectiveStartDate%20asc&%24filter=Schedule%2FEffectiveStartDate%20ne%20null",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 60,
      },
    },
  ],
]
`;

exports[`Events should get all recent events using the plugin 1`] = `
Array [
  Object {
    "id": 123,
    "name": "some event",
  },
]
`;

exports[`Events should get all recent events using the plugin 2`] = `
Array [
  Array [
    "Apollos/GetEventItemOccurencesByCalendarId?id=1&loadAttributes=expanded&%24expand=Schedule&%24orderby=Schedule%2FEffectiveStartDate%20asc&%24filter=Schedule%2FEffectiveStartDate%20ne%20null",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 60,
      },
    },
  ],
]
`;

exports[`Events should get by Campus 1`] = `
Array [
  Object {
    "id": 123,
    "name": "some event",
  },
]
`;

exports[`Events should get by Campus 2`] = `
Array [
  Array [
    "EventItemOccurrences?loadAttributes=expanded&%24expand=Schedule&%24orderby=Schedule%2FEffectiveStartDate%20asc&%24filter=%28Schedule%2FEffectiveStartDate%20ne%20null%29%20or%20%28CampusId%20eq%20456%29",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 60,
      },
    },
  ],
]
`;

exports[`Events should get from id 1`] = `null`;

exports[`Events should get from id 2`] = `
Array [
  Array [
    "EventItemOccurrences?loadAttributes=expanded&%24filter=Id%20eq%20123&%24expand=Schedule&%24top=1",
    Object {},
    Object {},
  ],
]
`;

exports[`Events should return a description based on an EventItem 1`] = `
Array [
  Array [
    "EventItems/123?loadAttributes=expanded",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 60,
      },
    },
  ],
]
`;

exports[`Events should return a name based on an EventItem 1`] = `
Array [
  Array [
    "EventItems/123?loadAttributes=expanded",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 60,
      },
    },
  ],
]
`;

exports[`Events should return an image based on an EventItem 1`] = `
Object {
  "sources": Array [
    Object {
      "uri": "https://example.com",
    },
  ],
}
`;

exports[`Events should return an image based on an EventItem 2`] = `
Array [
  Array [
    "EventItems/123?loadAttributes=expanded",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 60,
      },
    },
  ],
]
`;

exports[`Events should return an image based on an EventItem 3`] = `
Array [
  Array [
    Object {
      "id": undefined,
    },
  ],
]
`;

exports[`Events should return start and end based on a schedule 1`] = `
Object {
  "end": "2013-05-01T23:00:00Z",
  "start": "2013-05-01T22:00:00Z",
}
`;
